sentry-android-core/src/main/java/io/sentry/android/core/ViewHierarchyEventProcessor.java

```diff
@@ -0,0 +1,137 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import android.view.View;
+import android.view.ViewGroup;
+import io.sentry.Attachment;
+import io.sentry.EventProcessor;
+import io.sentry.Hint;
+import io.sentry.SentryEvent;
+import io.sentry.SentryLevel;
+import io.sentry.android.core.internal.gestures.ViewUtils;
+import io.sentry.protocol.ViewHierarchy;
+import io.sentry.protocol.ViewHierarchyNode;
+import io.sentry.util.Objects;
+import java.io.BufferedWriter;
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** ViewHierarchyEventProcessor responsible for taking a snapshot of the current view hierarchy. */
+@ApiStatus.Internal
+public final class ViewHierarchyEventProcessor implements EventProcessor {
+
+  private final @NotNull SentryAndroidOptions options;
+
+  public ViewHierarchyEventProcessor(final @NotNull SentryAndroidOptions options) {
+    this.options = Objects.requireNonNull(options, "SentryAndroidOptions is required");
+  }
+
+  @Override
+  public @NotNull SentryEvent process(final @NotNull SentryEvent event, @NotNull Hint hint) {
+    if (!event.isErrored()) {
+      return event;
+    }
+
+    if (!options.isAttachViewHierarchy()) {
+      this.options.getLogger().log(SentryLevel.DEBUG, "attachViewHierarchy is disabled.");
+      return event;
+    }
+
+    final @Nullable Activity activity = CurrentActivityHolder.getInstance().getActivity();
+    if (activity == null) {
+      return event;
+    }
+
+    try {
+      final @NotNull ViewHierarchy viewHierarchy = snapshotViewHierarchy(activity);
+      @SuppressWarnings("CharsetObjectCanBeUsed")
+      final Charset UTF8 = Charset.forName("UTF-8");
+
+      try (final ByteArrayOutputStream stream = new ByteArrayOutputStream();
+          final Writer writer = new BufferedWriter(new OutputStreamWriter(stream, UTF8))) {
+
+        options.getSerializer().serialize(viewHierarchy, writer);
+
+        final Attachment attachment = Attachment.fromViewHierarchy(stream.toByteArray());
+        hint.setViewHierarchy(attachment);
+      }
+    } catch (Throwable t) {
+      options.getLogger().log(SentryLevel.ERROR, "Could not snapshot ViewHierarchy", t);
+    }
+
+    return event;
+  }
+
+  @NotNull
+  public static ViewHierarchy snapshotViewHierarchy(Activity activity) {
+    final List<ViewHierarchyNode> windows = new ArrayList<>();
+    final ViewHierarchy viewHierarchy = new ViewHierarchy("android_view_system", windows);
+
+    final @Nullable View decorView = activity.getWindow().peekDecorView();
```

Comment:
M: `activity.getWindow()` can return null, a null check is cheaper than throwing the exception.